### PR TITLE
fix: rate limit admin endpoints

### PR DESCRIPTION
**Pull request by gagantrivedi**
## TODO
- [x] https://github.com/Flagsmith/flagsmith/pull/2703/files#r1361535311

Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Rate limit admin endpoints 

## How did you test this code?

Adds unit test case

### CODE REVIEW

**Comment by matthewelwell** [comment]
Hmm... yeah, I can't work out what's going on here, but if I add a breakpoint [here](https://github.com/encode/django-rest-framework/blob/master/rest_framework/throttling.py#L131), I can see that the the throttle rate for the `"user"` scope is being overwritten as `1/minute`. I can't quite work out why...

```
(Pdb) self.cache
<django.core.cache.backends.db.DatabaseCache object at 0x10c094410>
(Pdb) self.get_rate()
'1/minute'
(Pdb) from django.conf import settings
(Pdb) settings.REST_FRAMEWORK
{'DEFAULT_PERMISSION_CLASSES': ['rest_framework.permissions.IsAuthenticated'], 'DEFAULT_AUTHENTICATION_CLASSES': ('rest_framework.authentication.TokenAuthentication', 'api_keys.authentication.MasterAPIKeyAuthentication'), 'PAGE_SIZE': 10, 'UNICODE_JSON': False, 'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination', 'DEFAULT_THROTTLE_CLASSES': ['core.throttling.DBBackedUserRateThrottle'], 'DEFAULT_THROTTLE_RATES': {'login': '1/sec', 'mfa_code': '5/min', 'invite': '10/min', 'signup': '1000/min', 'user': '1/minute'}, 'DEFAULT_FILTER_BACKENDS': ['django_filters.rest_framework.DjangoFilterBackend'], 'DEFAULT_RENDERER_CLASSES': ['util.renderers.PydanticJSONRenderer', 'rest_framework.renderers.BrowsableAPIRenderer']}
(Pdb) settings.USER_THROTTLE_RATE
'10000/second'
```

**Comment by matthewelwell** [comment]
Ah, it's because our tests are not stateless! By setting this value [here](https://github.com/Flagsmith/flagsmith/pull/2703/files#diff-16b225283ebfee2d02fac7812f66a0b1c36b5ba0150ef5d576e40d660907b5e1R998), all subsequent tests receive that value.

File: api/app/settings/common.py [hunk 1]
```diff
@@ -217,6 +217,7 @@
 
 LOGIN_THROTTLE_RATE = env("LOGIN_THROTTLE_RATE", "20/min")
 SIGNUP_THROTTLE_RATE = env("SIGNUP_THROTTLE_RATE", "10000/min")
+USER_THROTTLE_RATE = env("USER_THROTTLE_RATE", "10/second")
```
Comment (gagantrivedi) [inline api/app/settings/common.py]: @matthewelwell we need to decide a sensible limit here;

Comment (matthewelwell) [reply api/app/settings/common.py]: Hmm... yeah, good question. 10/second seems reasonable to me for now tbh.

Comment (matthewelwell) [reply api/app/settings/common.py]: Maybe 500/min is better?

Comment (gagantrivedi) [reply api/app/settings/common.py]: Yeah, think we should start with a higher limit

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**